src/Sema.zig

```diff
@@ -17398,6 +17398,21 @@ fn resolvePeerTypes(
     const target = sema.mod.getTarget();
 
     var chosen = instructions[0];
+    var err_set_ty: ?Type = blk: {
+        const chosen_ty = sema.typeOf(chosen);
+
+        // TODO: is this the right handling of generic poison?
+        if (chosen_ty.tag() == .generic_poison or chosen_ty.zigTypeTag() != .ErrorSet)
```

Comment:
pass it up the stack:

```suggestion
        const chosen_ty_tag = try chosen_ty.zigTypeTagOrPoison();
        if (chosen_ty_tag != .ErrorSet)
```

This mechanism is (at least for now) how we obtain a function type for an un-instantiated generic function. We try to obtain parameter types, but it might not be possible if the type depends on something that is only available in the instantiation. Example:

```zig
fn foo(T: type, x: T, other: i32) void {}
```

In this example when we look at `@TypeOf(foo)`, the parameter `x` will cause generic poison, but `other` will return the type `i32` for that parameter.

Hope that helps clear things up.

Reply:
Thanks, done!